stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,661 @@
+    export *
+    explicit module sys_unistd {
+      header "sys/unistd.h"
+      export *
+    }
+  }
+  module utime {
+    header "utime.h"
+    export *
+  }
+  export *
+}
+
+module _dl [system] {
+  // This module groups all dl* based operations
+  // into one module.
+  module dlfcn {
+    header "dlfcn.h"
+    export *
+  }
+  module _link {
+    header "link.h"
+    export *
+  }
+  export *
+}
+
+module _error [system] {
+  module err {
+    header "err.h"
+    export *
+  }
+  module error {
+    header "error.h"
+    export *
+  }
+  export *
+}
+
+module _execinfo [system] {
+  header "execinfo.h"
+  export *
+}
+
+module _features [system] {
+  header "features.h"
+  export *
+}
+
+module _ftw [system] {
+  header "ftw.h"
+  export *
+}
+
+module _getopt [system] {
+  header "getopt.h"
+  export *
+}
+
+module _grp [system] {
+  header "grp.h"
+  export *
+}
+
+module _iconv [system] {
+  header "iconv.h"
+  export *
+}
+
+module _inet [system] {
+  // This module groups headers related to inet
+  // and networking.
+  module ifaddrs {
+    header "ifaddrs.h"
+    export *
+  }
+  module netdb {
+    header "netdb.h"
+    export *
+  }
+  module arpa_inet {
+    header "arpa/inet.h"
+    export *
+  }
+  module netinet_in {
+    header "netinet/in.h"
+    export *
+  }
+  module netinet_in6 {
+    header "netinet/in6.h"
+    export *
+  }
+  export *
+}
+
+module _jni [system] {
+  header "jni.h"
+  export *
+}
+
+module _langinfo [system] {
+  header "langinfo.h"
+  export *
+}
+
+module _pthread [system] {
+  header "pthread.h"
+  export *
+}
+
+module _regex [system] {
+  header "regex.h"
+  export *
+}
+
+module _resolv [system] {
+  header "resolv.h"
+  export *
+}
+
+module _sched [system] {
+  header "sched.h"
+  export *
+}
+
+module _search [system] {
+  header "search.h"
+  export *
+}
+
+module _semaphore [system] {
+  header "semaphore.h"
+  export *
+}
+
+module _spawn [system] {
+  header "spawn.h"
+  export *
+}
+
+module _strings [system] {
+  header "strings.h"
+  export *
+}
+
+module _sys [system] {
+  module syscall {
+    header "syscall.h"
+    export *
+  }
+  module sysexits {
+    header "sysexits.h"
+    export *
+  }
+  module syslog {
+    header "syslog.h"
+    export *
+  }
+  export *
+}
+
+module _tar [system] {
+  header "tar.h"
+  export *
+}
+
+module _termio [system] {
+  module termio {
+    header "termio.h"
+    export *
+  }
+  module termios {
+    header "termios.h"
+    export *
+  }
+  export *
+}
+
+module _threads [system] {
+  header "threads.h"
+  export *
+  explicit module threads_inlines {
+    header "bits/threads_inlines.h"
+    export *
+  }
+}
+
+module _uconfig_local [system] {
+  header "uconfig_local.h"
+  export *
+}
+
+module _ucontext [system] {
+  header "ucontext.h"
+  export *
+}
+
+module _wait [system] {
+  header "wait.h"
+  export *
+  explicit module sys_wait {
+    header "sys/wait.h"
+    export *
+  }
+}
+
+module _wchar_uchar [system] {
+  module uchar {
```

Comment:
uchar.h and wchar.h need to be independent modules because they each have c++ overrides